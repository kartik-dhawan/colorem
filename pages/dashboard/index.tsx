import { Box } from "@mui/material"
import SubNav from "../../components/common/Navbar/SubNav"
import FileSection from "../../components/FileSection"
import Essentials from "../../components/Essentials"
import FeedbackAndTeam from "../../components/FeedbackAndTeam"
import { client } from "../../utils/contentful/config"
import { ContentfulType } from "../../utils/interfaces"
import { useEffect } from "react"
import { useDispatch } from "react-redux"
import { updateContent } from "../../redux/slices/contentSlice"

export const getStaticProps = async () => {
  const response = await client.getEntries({ content_type: "coloremDashboard" }) // eslint-disable-line

  return {
    props: {
      data: response?.items[0]?.fields,
    },
  }
}

const Dashboard = ({ data }: ContentfulType) => {
  const dispatch = useDispatch()

  useEffect(() => {
    dispatch(updateContent(data))
  }, [data])

  return (
    <Box>
      <SubNav content={data} />
      <FileSection />
      <FeedbackAndTeam />
      <Essentials />
    </Box>
  )
}

export default Dashboard
