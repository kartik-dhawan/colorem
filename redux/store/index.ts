import { configureStore } from "@reduxjs/toolkit"
import countSlice from "../slices/countSlice"

const store = configureStore({
  reducer: {
    countSlice,
  },
})

export default store
