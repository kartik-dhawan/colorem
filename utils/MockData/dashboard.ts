export const contentfulDefaultData = {
  subnavHeader: "PERMUTATIONS WITH COLORS AND MORE.",
  browseCardContent: "Because your project needs more colors.",
  feedbackTitle: "WORKING ON YOUR FEEDBACKS!",
  commentFormatText: "sent a feedback",
  feedbackFormatText: "sent a comment",
  teamSectionBody:
    "The super fast color palettes generator! Create the perfect color palette or get inspired by thousands of beautiful color schemes. Explore trending palettes now & start creating.",
  essentialItemsText: [
    {
      id: 1,
      body: "Generate a random palette for your web design. Save it if you like it else hop on to the next.",
    },
    {
      id: 2,
      body: "Bored of browsing palettes? Make your own with a combination of your creativity and our dataset.",
    },
    {
      id: 3,
      body: "We have pre-built templates for color gradients; ready for you to use & make your website radiant.",
    },
  ],
  footerReferenceText:
    "This web app is only made as a project and in complete reference to the following resources...",
  footerDevelopedByText: "Coded & developed completely by ",
  developedBy: "Kartik Dhawan",
  references: [
    {
      type: "UI/UX Design",
      title: "Halo Lab",
      source:
        "https://dribbble.com/shots/18365166-Web-site-design-landing-page-home-page-ui",
      sourceName: "Dribbble",
    },
    {
      type: "Web Application",
      title: "Coolers",
      source: "https://coolors.co/",
      sourceName: "Coolers",
    },
    {
      type: "API",
      title: "Colormind",
      source: "http://colormind.io/api-access/",
      sourceName: "Colormind API",
    },
  ],
  teamMembers: [
    {
      metadata: { tags: [] },
      sys: {
        space: {
          sys: { type: "Link", linkType: "Space", id: "1v98waf66sfj" },
        },
        id: "5L8OizpXMHyrYgM5ao6NTx",
        type: "Asset",
        createdAt: "2022-12-08T22:01:54.500Z",
        updatedAt: "2022-12-08T22:01:54.500Z",
        environment: {
          sys: { id: "master", type: "Link", linkType: "Environment" },
        },
        revision: 1,
        locale: "en-US",
      },
      fields: {
        title: "Kartik Dhawan",
        description: "Avatar of Kartik Dhawan",
        file: {
          url: "//images.ctfassets.net/1v98waf66sfj/5L8OizpXMHyrYgM5ao6NTx/013c075eacd65ad632e43442e0b620c6/user1.JPG",
          details: { size: 34091, image: { width: 400, height: 400 } },
          fileName: "user1.JPG",
          contentType: "image/jpeg",
        },
      },
    },
    {
      metadata: { tags: [] },
      sys: {
        space: {
          sys: { type: "Link", linkType: "Space", id: "1v98waf66sfj" },
        },
        id: "3gNCsKWDkm9rWiNkurM5Te",
        type: "Asset",
        createdAt: "2022-12-08T22:02:45.218Z",
        updatedAt: "2022-12-08T22:02:45.218Z",
        environment: {
          sys: { id: "master", type: "Link", linkType: "Environment" },
        },
        revision: 1,
        locale: "en-US",
      },
      fields: {
        title: "Gretchen Morgan",
        description: "Avatar of Gretchen Morgan",
        file: {
          url: "//images.ctfassets.net/1v98waf66sfj/3gNCsKWDkm9rWiNkurM5Te/579d621f150b8f9ebf301e8a27fce0e6/user2.PNG",
          details: { size: 411188, image: { width: 500, height: 500 } },
          fileName: "user2.PNG",
          contentType: "image/png",
        },
      },
    },
    {
      metadata: { tags: [] },
      sys: {
        space: {
          sys: { type: "Link", linkType: "Space", id: "1v98waf66sfj" },
        },
        id: "43okVLERYKi8ibgIVdWDpb",
        type: "Asset",
        createdAt: "2022-12-08T22:03:17.269Z",
        updatedAt: "2022-12-08T22:03:17.269Z",
        environment: {
          sys: { id: "master", type: "Link", linkType: "Environment" },
        },
        revision: 1,
        locale: "en-US",
      },
      fields: {
        title: "Dunder Mifler",
        description: "Avatar of Dunder Mifler",
        file: {
          url: "//images.ctfassets.net/1v98waf66sfj/43okVLERYKi8ibgIVdWDpb/6c4334b2e82b16d038df934d114ee16d/user3.JPG",
          details: { size: 64676, image: { width: 623, height: 615 } },
          fileName: "user3.JPG",
          contentType: "image/jpeg",
        },
      },
    },
  ],
  sideNavigationItems: [
    { id: 1, body: "Go back to landing page.", link: "/", title: "Home" },
    {
      id: 2,
      body: "Browse through the color palettes for web-aesthetic.",
      link: "/palettes",
      title: "Palettes",
    },
    {
      id: 3,
      body: "Browse pre-built templates for color gradients.",
      link: "/gradients",
      title: "UI Gradiants",
    },
    {
      id: 4,
      body: "Go deep into the technical side of the project",
      link: "/about/more",
      title: "More on the project",
    },
  ],
  quickLinks: [
    { id: 1, path: "/about", title: "About" },
    { id: 2, path: "/about/more", title: "More" },
    { id: 3, path: "/palettes/create", title: "Create a palette" },
  ],
  paletteSectionInstructor: [
    {
      id: 1,
      icon: "spacebar",
      instruction:
        "Hit spacebar or next palette icon next to 'Like' button to go to next palette.",
    },
    {
      id: 2,
      icon: "tap",
      instruction:
        "Click/tap on the hex-code written in any individual palette to copy it",
    },
  ],
  gradientSectionInstructor: [
    {
      id: 1,
      icon: "tap",
      instruction: "Hover/Tap on the gradient to view available actions.",
    },
    {
      id: 2,
      icon: "info",
      instruction:
        "Click on 'info' icon to get more information & code of the gradient.",
    },
  ],
  customErrorPagesData: {
    "404": {
      id: 1,
      statusBody: "The page you're looking for doesn't exist.",
      statusCode: "404",
      statusTitle: "Uncharted waters!",
    },
    redirectBtnText: "Get back to dashboard?",
  },
  defaultMetaImage: {
    metadata: { tags: [] },
    sys: {
      space: {
        sys: { type: "Link", linkType: "Space", id: "1v98waf66sfj" },
      },
      id: "3AZSHDIcX2l5wtugFxRmWn",
      type: "Asset",
      createdAt: "2023-01-31T19:35:42.698Z",
      updatedAt: "2023-01-31T20:21:25.407Z",
      environment: {
        sys: { id: "master", type: "Link", linkType: "Environment" },
      },
      revision: 5,
      locale: "en-US",
    },
    fields: {
      title: "Home Meta Image",
      description: "",
      file: {
        url: "//images.ctfassets.net/1v98waf66sfj/3AZSHDIcX2l5wtugFxRmWn/0d56e19f5f34ca6dd7a1ff453cefdcf4/homeMetaImage.webp",
        details: { size: 13532, image: { width: 1090, height: 1090 } },
        fileName: "homeMetaImage.webp",
        contentType: "image/webp",
      },
    },
  },
  aboutPageNavItems: [
    { id: 1, route: "project", title: "About the project" },
    { id: 2, route: "dev", title: "Behind the desk" },
    { id: 3, route: "tech", title: "Tech nuggets" },
    { id: 4, route: "team", title: "The colorem team" },
    { id: 5, route: "more", title: "More" },
  ],
  aboutPageDescription:
    "A little too much information on this page. Make the tone warmer and get you reading glasses to read more about the project.\n",
  portfolioUrl: "https://kartikdhawan.vercel.app",
}
