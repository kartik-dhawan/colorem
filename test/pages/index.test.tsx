import Home from "../../pages"
import { Provider } from "react-redux"
import store from "../../redux/store"
import { contentfulDefaultData } from "../../utils/MockData/dashboard"
import renderer from "react-test-renderer"
import { render } from "@testing-library/react"
import "@testing-library/jest-dom"

describe("Home page", () => {
  it("-- should match snapshot", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Home contentData={contentfulDefaultData} />
        </Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("-- should render", () => {
    const view = render(
      <Provider store={store}>
        <Home contentData={contentfulDefaultData} />
      </Provider>
    )

    const homeTitle = view.getByText("Colorem")

    expect(homeTitle).toBeInTheDocument()
  })
})
