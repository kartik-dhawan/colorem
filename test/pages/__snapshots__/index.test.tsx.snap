// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page -- should match snapshot 1`] = `
<div
  className="indexWrapper"
  data-testid="indexWrapper"
>
  <main
    className="indexMain"
  >
    <a
      className="indexHeader className"
      href="/dashboard"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Colorem
    </a>
    <a
      className="indexLink"
      href="/dashboard"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Start creating
    </a>
  </main>
</div>
`;
